src/components/ApplicationTimer.js

```diff
@@ -0,0 +1,55 @@
+import { useState, useEffect } from 'react'
+import { calculateTimeLeft, formatDoubleDigitTime } from '../utils/timer'
+import styles from '../css/CountdownTimer.module.css'
+
+const getTime = dueDate => {
+  const timeLeft = calculateTimeLeft(dueDate)
+
+  if (timeLeft) {
+    timeLeft.hours = formatDoubleDigitTime(timeLeft.hours)
+    timeLeft.minutes = formatDoubleDigitTime(timeLeft.minutes)
+    timeLeft.seconds = formatDoubleDigitTime(timeLeft.seconds)
+  }
+
+  return timeLeft
+}
+
+const ApplicationTimer = () => {
+  const dueDate = '01/25/2021'
+  const [timeLeft, setTimeLeft] = useState(getTime(dueDate))
+
+  useEffect(() => {
```

Comment:
if you give the useEffect an empty dep array (`useEffect( ()=>{}, [])`, so it only runs once on creation) you can use a setInterval and clearInterval instead. This will be more accurate to real time 1s because this timeout does this:
render -> wait 1 sec -> state update -> rerender -> wait 1 sec -> state update -> ....
in the event of long render and update times, it might wait something like 1.15s before updating. But if you use the interval, it will call a state update every second no matter if react is rendering or anything.